tests/framework/db/QueryTest.php

```diff
@@ -166,6 +167,9 @@ public function testOrder()
 
         $query->addOrderBy('age ASC, company DESC');
         $this->assertEquals(['team' => SORT_ASC, 'company' => SORT_DESC, 'age' => SORT_ASC], $query->orderBy);
+
+        $query->orderBy(new Expression('CASE WHEN x = y THEN 0 ELSE 1 END DESC'));
```

Comment:
this will internally be cased to a string and not fail as it does not contain a `,` if you add an `Expression` that contains a comma, it should fail.